lib/system.nim

```diff
@@ -1269,6 +1253,28 @@ proc add*(x: var string, y: string) {.magic: "AppendStrStr", noSideEffect.}
   ##   tmp.add("cd")
   ##   assert(tmp == "abcd")
 
+proc len*[TOpenArray: openArray|varargs](x: TOpenArray): int {.
+  magic: "LengthOpenArray", noSideEffect.}
+proc len*(x: string): int {.magic: "LengthStr", noSideEffect.}
+proc len*(x: cstring): int {.magic: "LengthStr", noSideEffect.}
+proc len*[I, T](x: array[I, T]): int {.magic: "LengthArray", noSideEffect.}
+proc len*[T](x: seq[T]): int {.magic: "LengthSeq", noSideEffect.}
+  ## returns the length of an array, an openarray, a sequence or a string.
+  ## This is roughly the same as ``high(T)-low(T)+1``, but its resulting type is
+  ## always an int.
+  ##
+  ## .. code-block:: nim
+  ##  var arr = [1,1,1,1,1]
+  ##  len(arr) #=> 5
+  ##  for i in 0..<arr.len:
+  ##    echo arr[i] #=> 1,1,1,1,1
+
+proc len*[T: Ordinal](x: Slice[T]): int {.inline, noSideEffect.} =
+  ## returns the length of an ordinal slice
+  result = ord(x.b) - ord(x.a) + 1
+  if result < 0: result = 0
```

Comment:
(for i in 5..2)  gives zero iterations not 4, imo len should be consistent with current behavior